private/Message.hpp

```diff
@@ -15,23 +15,32 @@
 #include <string>
 #include <vector>
 
-class IrcMessage
+class Message
 {
+public:
+	// Constructors
+	Message(std::string const &msg);
+	
+	// Destructor
+	~Message(void);
+
+	// Accessors
+	std::string const        &get_prefix() const;
+	std::string const        &get_command() const;
+	std::vector<std::string> &get_params();
+
+	// Setters
+	void set_prefix(std::string const &prefix);
+	void set_command(std::string const &command);
+	void set_params(std::vector<std::string> const &params);
+
+	// Methods
+	void add_param(std::string const &param);
```

Comment:
If it's used only internally for parsing, it should be private, else I think it's useless.